src/library/trackcollection.h

```diff
@@ -66,6 +66,10 @@ class TrackCollection : public QObject,
 
     void relocateDirectory(QString oldDir, QString newDir);
 
+    // This function returns a track ID of all file in the list not already visible,
+    // it adds and unhides the tracks as well.
+    QList<TrackId> getAndEnsureTrackIds(const QList<QFileInfo>& files, bool addMissingTracks);
```

Comment:
The name of this function has to be improved if you need to add detailed comments upon every usage. Also the name "get" doesn't fit if it is no read-only or idempotent.

Reply:
This is actually the third name of this function. A strong hint that it does to much at once. :-/  
Unfortunately this is required for a quick execution. 
I was lately fine with this name, but I am happy to change it.

Ideas: 
createOrGetTrackIds() 
addMissingTracksAndGetAllTrackIds()